File: 1-conditions/bisextile.exs
# programme calculant années bisextiles
defmodule Annee do

  def is_bisextile?(annee) do
    cond do
      rem(annee, 4)   != 0 -> false
      rem(annee, 100) != 0 -> true
      rem(annee, 400) == 0 -> true
      true                 -> false
    end
  end
end

#main
annee = IO.gets "Entrez une année : "
annee = annee |> String.trim |> String.to_integer

case Annee.is_bisextile?(annee) do
  true  -> IO.puts "#{annee} est une année bisextile."
  false -> IO.puts "#{annee} n'est pas une année bisextile."
  _     -> IO.puts "Tu as encore du debug à faire mon gars…"
end

File: 1-conditions/bisextile.py
# programme calculant années bisextiles
annee = input("Enrtrez une année : ")
annee = int(annee) # convertir en entier

if (annee % 4 ) != 0:
  bisextile = False
elif (annee % 100) != 0:
  bisextile = True
elif (annee % 400) == 0:
  bisextile = True
else:
  bisextile = False

# on affiche le résultat
if bisextile == True:
  print(annee, " est une année bisextile.")
else:
  print(annee, " n'est pas une année bisextile")